### PR TITLE
annette

### PR DESCRIPTION
**Pull request by annettefo**
To brendan

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+});
+//EDIT
+app.get('/:id/edit/', function(req, res){
+  Mongoose.find({ _id: req.params.id }, function(err, response) {
+    if (err) { console.log(err); }
+    res.render('edit', { mongoose: response[0] });
+  })
+});
+// UPDATE
+app.post('/:id', function(req, res){
```
Comment (brendan-codes) [inline server.js:79]: This line is a little scary. Our route here says that we'll accept literally any HTTP request. We can run `/magicalunicorns` and we can run `/new` through this route, and it's going to take that ID straight to the database with no validation.

Our recommendation is to make the route a little more specific, example: `/mongoose/:id`

File: server.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+app.get('/', (req, res) => {
+  db.collection('mongooses').find().toArray((err, result) => {
+    if (err) return console.log(err)
+    // renders index.ejs
+    console.log(result);
+    res.render('index.ejs', {mongooses: result})
+  })
+})
+// DESTROY THE MONGOOSE
+app.post('/:id/delete', function(req, res){
```
Comment (brendan-codes) [inline server.js:65]: It's a good practice to never start routes with symbols. Flipping this around so it reads `/delete/:id` would be preferred. We just don't want conflict with our other routes.

File: server.js [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  var MongooseSchema = new mongoose.Schema({
+   name: String,
+   info: String,
+   comment: String
+  }, { versionKey: false });
+  mongoose.model('Mongoose', MongooseSchema); // We are setting this Schema in our Models as 'User'
+  var Mongoose = mongoose.model('Mongoose') // We are retrieving this Schema from our Models, named 'User'
+// Routes
+// Root Request
+// app.get('/', function(req, res) {
```
Comment (brendan-codes) [inline server.js:35]: Clean out commented code that's no longer needed when possible, it just makes our code base more manageable and readable.

File: server.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
+var mongoose = require('mongoose');
+// Require the Express Module
+var express = require('express');
+// Create an Express App
+var app = express();
+// Require body-parser (to receive post data from clients)
+var bodyParser = require('body-parser');
+var db = mongoose.connection;
```
Comment (brendan-codes) [inline server.js:8]: This is a neat little way to talk to our DB directly, however it's still recommended to use our model variables whenever possible so we can leverage all of mongoose's great features!

File: views/index.ejs [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+            <tr>
+              <th>Name</th>
+              <th>Information</th>
+              <th> Edit </th>
+              <th> View </th>
+              <th> Delete </th>
+            </tr>
+          </thead>
+          <tbody>
+          <% for (mongoose in mongooses) { %>
```
Comment (brendan-codes) [inline views/index.ejs:47]: `for (var mongoose in mongooses) {` Always use the var keyword, even in loops!

**Review by brendan-codes — CHANGES_REQUESTED** [review]
Great job, code is pretty clean. I recommend just clearing out your experimental code and comments in the future, especially for production level codebases. I recommend keeping your route names RESTful as well.

**Comment by annettefo** [comment]
thanks!

File: server.js (hunk 4 above)
Comment (annettefo) [reply server.js:8]: Thank you!

File: views/index.ejs (hunk 5 above)
Comment (annettefo) [reply views/index.ejs:47]: Noted!

**Pull request closed without merging**